addons/website_event/controllers/main.py

```diff
@@ -91,6 +112,18 @@ def events(self, page=1, slug_tags=None, **searches):
         events = event_details.get('results', Event)
         events = events[(page - 1) * step:page * step]
 
+        country = None
+        event_location = website.is_view_active('website_event.event_location')
+        if event_location:
+            country = self._get_default_country_search(events)
+
+        if not event_location and not searches.get('country'):
+            searches['country'] = 'all'
+        elif event_location and not searches.get('country'):
+            return request.redirect(f'{request.httprequest.full_path}&country={country}')
+        elif not event_location and searches.get('country') and searches.get('country') != 'all':
+            return request.redirect('/event')
```

Comment:
Suggestion:

```python
  if not searches.get('country'):
      if country and country != 'all' and event_location:
        return request.redirect(f'{request.httprequest.full_path}&country={country}')
      searches['country'] = 'all'
  elif searches.get('country') != 'all' and not event_location:
      return request.redirect('/event')
```

Not sure about the `elif` : it seems like we do not want to allow the user to be able to manually filter if the option is not enabled? That seems restrictive, not sure we should keep it unless we really want that. But maybe I'm missing something :)